src/ComposerGpgVerify/Verify.php

```diff
@@ -0,0 +1,213 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ComposerGpgVerify;
+
+use Composer\Composer;
+use Composer\Config;
+use Composer\EventDispatcher\EventSubscriberInterface;
+use Composer\IO\IOInterface;
+use Composer\Plugin\PluginInterface;
+use Composer\Script\Event;
+use Composer\Script\ScriptEvents;
+
+final class Verify implements PluginInterface, EventSubscriberInterface
+{
+    private function __construct()
+    {
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public static function getSubscribedEvents() : array
+    {
+        return [
+            ScriptEvents::PRE_AUTOLOAD_DUMP => 'verify',
+        ];
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @codeCoverageIgnore
+     */
+    public function activate(Composer $composer, IOInterface $io)
+    {
+        // Nothing to do here, as all features are provided through event listeners
+    }
+
+    /**
+     * @param Event $composerEvent
+     * @throws \LogicException
+     * @throws \RuntimeException
+     */
+    public static function verify(Event $composerEvent) : void
+    {
+        $composer         = $composerEvent->getComposer();
+        $config           = $composer->getConfig();
+
+        self::assertSourceInstallation($config);
+
+        $vendorDir = $config->get('vendor-dir');
+
+        // @todo that `$vendorDir` may include special chars
+        /* @var $vendorDirs \GlobIterator|\SplFileInfo[]*/
+        $vendorDirs = new \GlobIterator(
+            $vendorDir . '/*/*',
+            \FilesystemIterator::KEY_AS_PATHNAME | \FilesystemIterator::FOLLOW_SYMLINKS | \FilesystemIterator::SKIP_DOTS | \FilesystemIterator::CURRENT_AS_FILEINFO
+        );
+
+        $packages = [];
+
+        foreach ($vendorDirs as $vendorDir) {
+            if (! $vendorDir->isDir()) {
+                continue;
+            }
+
+            $packageName = basename(dirname($vendorDir->getRealPath())) . '/' . $vendorDir->getBasename();
+
+            if (! is_dir($vendorDir->getRealPath() . '/.git')) {
+                $packages[$packageName] = [
+                    'git'       => false,
+                    'signed'    => false,
+                    'signature' => '',
+                    'verified'  => false,
+                ];
+
+                continue;
+            }
+
+            // because PHP is a moronic language, by-ref is everywhere in the standard library
+            $output = [];
+
+            // @TODO check check tags if the commit isn't signed
+            exec(
+                sprintf(
+                    'git --git-dir %s verify-commit --verbose HEAD 2>&1',
+                    escapeshellarg($vendorDir->getRealPath() . '/.git')
```

Comment:
escapeshellarg is locale-dependent, some weird thing can happen: https://github.com/symfony/symfony/commit/3779f3fbb98db5fd1647fa8e2af84a05adb1c7a0